### PR TITLE
Support symlinked alternate run directories (cylc 7).

### PR DESCRIPTION
**Pull request by hjoliver**
7.8.x back-port of #2935

Addresses #2797.
 
**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (functional).
- [x] Includes an appropriate entry in the release change log `CHANGES.md`.
- [x] Includes an appropriate documentation update.

### CODE REVIEW

**Comment by hjoliver** [comment]
I've finally finished this off.  Ready for a quick second review (any volunteers?).  See explanation in description of #2935.

**Comment by matthewrmshin** [comment]
@wxtim please review 2.

**Comment by hjoliver** [comment]
Back-ported registration test fixes https://github.com/cylc/cylc-flow/pull/2935#issuecomment-526440821

File: tests/registration/00-simple.t [hunk 1]
```diff
@@ -14,119 +14,213 @@
...
 # Test suite registration
 
+export RND_SUITE_NAME
+export RND_SUITE_SOURCE
+export RND_SUITE_RUNDIR
+export CYLC_RUN_DIR
+
+CYLC_RUN_DIR="$(cylc get-global-config --print-run-dir)"
+
+function make_rnd_suite() {
```
Comment (wxtim) [inline tests/registration/00-simple.t]: I know that at the moment this is the only place this is used, but might these functions be better in test_header?

**Pull request merged**